aws/resource_aws_cognito_user_pool_client.go

```diff
@@ -53,6 +53,7 @@ func resourceAwsCognitoUserPoolClient() *schema.Resource {
 					ValidateFunc: validation.StringInSlice([]string{
 						cognitoidentityprovider.ExplicitAuthFlowsTypeAdminNoSrpAuth,
 						cognitoidentityprovider.ExplicitAuthFlowsTypeCustomAuthFlowOnly,
+						cognitoidentityprovider.ExplicitAuthFlowsTypeUserPasswordAuth,
```

Comment:
Could you add an acctest for that so that we ensure there won't be regression in the future?

Reply:
I thought about acctest before committing the change but not sure what to test actually. `USER_PASSWORD_AUTH` is a new type of `ExplicitAuthFlows`.

Reply:
My general opinion is if the SDK is providing the constant, that means AWS has added it to their acceptable values in their generated API model, and we don't need to provide any acceptance testing if its just updating a validation check. 😄

Reply:
I recall, when working on multiple Cognito ressources, that there was a specificity when using USER_PASSWORD_AUTH... hence my request :)
Adding the value in `testAccAWSCognitoUserPoolClientConfig_allFields` is fine as it covers the use-case.

Reply:
Added all values of `explicit_auth_flows` in acctest. I'm not clear about the user-case but hope to get your point of change.

Reply:
Cannot recall of the need for it right now... it was not that important, but will keep you updated if it appears to me again.